### PR TITLE
Add docs for VERSION

### PR DESCRIPTION
**Pull request by alexcb**
Signed-off-by: Alex Couture-Beil <alex@earthly.dev>

### CODE REVIEW

**Review by agbell — APPROVED** [review]
Looks good. 

One thing I would love to see, and maybe its more suited to the blog, is the thinking behind this flag. It's an example of us taking 'repeatable' builds seriously and could be a good excuse to talk about that.

**Comment by alexcb** [comment]
> One thing I would love to see, and maybe its more suited to the blog, is the thinking behind this flag. It's an example of us taking 'repeatable' builds seriously and could be a good excuse to talk about that.

During standup we talked a bit more about the environment override and #1125 and decided it's best to not even advertise this feature as it promotes non-repeatable builds. We'll leave the env variable override code in place; however will only use it as a dev feature, primarily to allow us to dogfood our features via the `./earthly` script in the root of the repo.

**Comment by dchw** [comment]
I like it. If we are planning on phasing these out over time adding the expiration date might be nice too.

**Review by dchw — APPROVED** [review]

File: docs/earthfile/earthfile.md [hunk 1]
```diff
@@ -584,6 +584,28 @@ For more information see the [multi-platform guide](../guides/multi-platform.md)
...
 
+## VERSION
+
+#### Synopsis
+
+* `VERSION [--use-copy-include-patterns] <version-number>`
+
+#### Description
+
+The command `VERSION` identifies which set of features to enable in Earthly while handling the corresponding Earthfile. The `VERSION` command is optional; however if specified, it must be the first command in the Earthfile.
```
Comment (vladaionescu) [inline docs/earthfile/earthfile.md:595]: We should also say that the `VERSION` command will become mandatory in a future version of Earthly.

File: docs/earthfile/earthfile.md [hunk 2]
```diff
@@ -584,6 +584,28 @@ For more information see the [multi-platform guide](../guides/multi-platform.md)
...
+| --- | --- | --- |
+| `0.5` | _initial functionality will be preserved_ |
+
+#### Options
+
+##### `--use-copy-include-patterns`
+
+Instructs Earthly to set buildkit's `include pattern` when executing `COPY` commands; without this feature, Earthly sends all files in the current build context except ignored files.
+Setting this option may increase the speed of COPY commands by forcing Earthly to only transfered files needed for the specific COPY, rather than transfering all files initially.
+This is disabled by default under `0.5`.
```
Comment (vladaionescu) [inline docs/earthfile/earthfile.md:607]: I think we should have a separate page with a table of feature flags like this one. We should make it clear that all feature flags are experimental.

File: docs/earthfile/earthfile.md [hunk 3]
```diff
@@ -584,6 +584,28 @@ For more information see the [multi-platform guide](../guides/multi-platform.md)
...
+
+| Version number | enabled features |
+| --- | --- | --- |
+| `0.5` | _initial functionality will be preserved_ |
+
+#### Options
+
+##### `--use-copy-include-patterns`
+
+Instructs Earthly to set buildkit's `include pattern` when executing `COPY` commands; without this feature, Earthly sends all files in the current build context except ignored files.
```
Comment (vladaionescu) [inline docs/earthfile/earthfile.md:605]: We should explain this from the perspective of the user: what is it that the user gains. We should assume that people don't know about BuildKit's internals here.

A better explanation could be "experimental faster COPY algorithm" or something along those lines.

**Review by vladaionescu — CHANGES_REQUESTED** [review]

File: docs/earthfile/earthfile.md (hunk 2 above)
Comment (grindernlm) [reply docs/earthfile/earthfile.md:607]: Ok

**Review by grindernlm — COMMENTED** [review]
True

File: docs/earthfile/earthfile.md (hunk 2 above)
Comment (alexcb) [reply docs/earthfile/earthfile.md:607]: If we list them on a second page, would we then want to change 

```
VERSION [--use-copy-include-patterns] <version-number>
```

to something like

```
VERSION [<feature-flags>...] <version-number>
```

so we wouldn't have to manage this list across two different files?

**Comment by alexcb** [comment]
I spun out the feature flags into a separate page:

image

Which then has the table listed here:

image

File: docs/SUMMARY.md [hunk 4]
```diff
@@ -40,6 +40,7 @@
 * [The `earthly` command](earthly-command/earthly-command.md)
 * [Configuration reference](earthly-config/earthly-config.md)
 * [CI integration](./ci-integration.md)
+* [Experimental features](experimental-features/features.md)
```
Comment (vladaionescu) [inline docs/SUMMARY.md:43]: Wondering if this should go under the Earthfile section rather?

Comment (alexcb) [reply docs/SUMMARY.md:43]: I was torn about that too, as some of the features aren't exactly related to Earthfiles, but I'm fine moving them there if we want to limit the number of top-level side links on the summary.

Comment (alexcb) [reply docs/SUMMARY.md:43]: I just moved it.

Comment (vladaionescu) [reply docs/SUMMARY.md:43]: Forgot to update SUMMARY too?

Comment (alexcb) [reply docs/SUMMARY.md]: good catch :)

File: docs/earthfile/earthfile.md [hunk 5]
```diff
@@ -584,6 +584,30 @@ For more information see the [multi-platform guide](../guides/multi-platform.md)
...
+| --- | --- | --- |
+| `0.5` | _initial functionality will be preserved_ |
+
+#### Options
+
+Individual features may be enabled by setting the corresponding feature flag.
+New features start off as experimental, which is why they are disabled by default.
+Once a feature reaches maturity, it will be enabled by default under a new version number.
+
+All features are described in [a corresponding table](features.md).
```
Comment (vladaionescu) [inline docs/earthfile/earthfile.md:609]: This link doesn't work: https://app.gitbook.com/@earthly/s/earthly-staging/~/revisions/-Mft5yKY3dlzLbngYGJX/v/version-docs/docs/earthfile#version

I wonder if it should be `./features.md` instead? Or maybe just git book weirdness that will fix itself with another run.

Comment (alexcb) [reply docs/earthfile/earthfile.md]: that definitely did the trick! thanks.

**Review by vladaionescu — APPROVED** [review]

**Pull request merged**